### PR TITLE
Hud tweaks

### PR DESCRIPTION
**Pull request by JMWTurner**
HUDs now update as soon as you enter a mob's body, which fixes a bug where re-entering your long-dead corpse would persist showing your health at 100%

Also simple animal slimes (obtained through pink slime reactions) also have the slime HUD now.

### CODE REVIEW

File: code/_onclick/hud/hud.dm [hunk 1]
```diff
@@ -218,6 +218,8 @@ var/global/obj/screen/clicker/catcher = new()
 		shade_hud()
 	else if(isslime(mymob))
 		slime_hud()
+	else if(istype(mymob, /mob/living/simple_animal/slime))
```
Comment (clusterfack) [inline code/_onclick/hud/hud.dm]: ...?

Comment (ghost) [reply code/_onclick/hud/hud.dm]: > Also simple animal slimes (obtained through pink slime reactions) also have the slime HUD now.

Comment (JMWTurner) [reply code/_onclick/hud/hud.dm]: I've pondered whether isslime should include them, but I thought it might break a few checks if it did.

Comment (clusterfack) [inline code/_onclick/hud/hud.dm]: Yeah I definitely think it should

**Comment by JMWTurner** [comment]
revision 1
- isslime now takes simple animal slimes into account

File: code/modules/mob/living/simple_animal/shade.dm [hunk 2]
```diff
@@ -94,6 +94,9 @@
 		return 0 //under effects of time magick
 	. = ..()
 
+	regular_hud_updates()
+
+/mob/living/simple_animal/shade/regular_hud_updates()
 	update_pull_icon() //why is this here?
```
Comment (ComicIronic) [inline code/modules/mob/living/simple_animal/shade.dm:100]: Could you remove this comment? Coders should be baffled no more by the pull hud update proc being inside the general hud update proc.

**Pull request merged**